src/test_workflow/perf_test/perf_single_node_cluster.py

```diff
@@ -0,0 +1,55 @@
+
+import os
+from contextlib import contextmanager
+
+from test_workflow.perf_test.perf_test_cluster import PerfTestCluster
+
+
+class PerfSingleNodeCluster(PerfTestCluster):
```

Comment:
How about if we can create one base class for the PerfCluster and pass the argument as `SingleNode` or `MultiNode` to avoid the duplication?

Reply:
The logic is totally different between single-node and multi-node cluster, that's why there are 2 child classes here.